[REQUEST]: Investigate utf8 issues with output: "Unknown code"

### Is your feature request related to a problem? Please describe

https://twitter.com/Cor3ntin/status/1502357281186402310?s=20&t=L1CnMe8pwqgIUOVetbIncA

https://godbolt.org/z/E6bbshGvv

getting bad unicode things in the output:

```
<source>:1:9: error: macro name must be an identifier
#define Unknown code<U+1F984> &
        ^
<source>:2:5: error: unexpected character <U+1F984>
int Unknown code<U+1F984>;
    ^~~~~~~~~
<source>:2:1: warning: declaration does not declare anything [-Wmissing-declarations]
int Unknown code<U+1F984>;
```

### Describe the solution you'd like

Unsure, but it seems confusing.

### Describe alternatives you've considered

na

### Additional context

Not applicable

@partouf this particular example fails in very old clang https://godbolt.org/z/eGhY4Tqb4  - but most importantly the output you get in a terminal is different

```foo.cpp:1:9: error: macro name must be an identifier
#define <U+1F984> &
        ^
foo.cpp:2:5: error: unexpected character <U+1F984>
int <U+1F984>;
    ^~~~~~~~~
foo.cpp:2:1: warning: declaration does not declare anything [-Wmissing-declarations]
int <U+1F984>;
^~~
1 warning and 2 errors generated.```

The difference is that "Unknown code" is injected in the diagnostic message somehow - 
I'm not confident at all it comes from CE but it certainly doesn't come from GCC

--- a/static/ansi-to-html.ts
+++ b/static/ansi-to-html.ts
@@ -167,6 +167,8 @@ function handleDisplay(stack: string[], _code: string | number, options: AnsiToH
         return codeMap[code]();
     } else if (4 < code && code < 7) {
         return pushTag(stack, 'blink');
+    } else if (code === 7) {
+        return '';
     } else if (29 < code && code < 38) {
         // @ts-ignore
         return pushForegroundColor(stack, options.colors[code - 30]);

--- a/test/ansi-to-html-tests.js
+++ b/test/ansi-to-html-tests.js
@@ -34,32 +34,38 @@ describe('ansi-to-html', () => {
     };
     it('Should leave non-ansi colours alone', () => {
         const filter = new Filter(filterOpts);
-        filter.toHtml('I am a boring old string')
-            .should.equal('I am a boring old string');
+        filter.toHtml('I am a boring old string').should.equal('I am a boring old string');
     });
     it('Should handle simple cases', () => {
         const filter = new Filter(filterOpts);
-        filter.toHtml('\x1B[38;5;99mTest')
-            .should.equal('<span style="color:#875fff">Test</span>');
+        filter.toHtml('\x1B[38;5;99mTest').should.equal('<span style="color:#875fff">Test</span>');
     });
     it('Should handle nasty edge cases', () => {
         const filter = new Filter(filterOpts);
         // See #1666, this used to cause catastrophic backtracking.
-        filter.toHtml('\x1B[38;5;9999999999999999999999999999999999999999999999999999999999999999999999999999999' +
-            '99999999999999999999"mTest').should.equal(
-            '5;9999999999999999999999999999999999999999999999999999999999999' +
-            '99999999999999999999999999999999999999"mTest');
+        filter
+            .toHtml(
+                '\x1B[38;5;9999999999999999999999999999999999999999999999999999999999999999999999999999999' +
+                    '99999999999999999999"mTest',
+            )
+            .should.equal(
+                '5;9999999999999999999999999999999999999999999999999999999999999' +
+                    '99999999999999999999999999999999999999"mTest',
+            );
     });
 
     // With thanks to https://github.com/rburns/ansi-to-html/pull/84/files
     it('renders xterm foreground 256 sequences', () => {
         const filter = new Filter(filterOpts);
-        filter.toHtml('\x1B[38;5;196mhello')
-            .should.equal('<span style="color:#ff0000">hello</span>');
+        filter.toHtml('\x1B[38;5;196mhello').should.equal('<span style="color:#ff0000">hello</span>');
     });
     it('renders xterm background 256 sequences', () => {
         const filter = new Filter(filterOpts);
-        filter.toHtml('\x1B[48;5;196mhello')
-            .should.equal('<span style="background-color:#ff0000">hello</span>');
+        filter.toHtml('\x1B[48;5;196mhello').should.equal('<span style="background-color:#ff0000">hello</span>');
+    });
+
+    it('should ignore reverse video', () => {
+        const filter = new Filter(filterOpts);
+        filter.toHtml('\x1B[7mhello').should.equal('hello');
     });
 });

